### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by cassella**
Fix typos in

Context.h, can-pass.h, cleanups.cpp, functionResolution.cpp,
chpl-init.cpp, and gpu-cuda.c,
buildingandrunning.rst, QUICKSTART.rst,
OS.chpl, Types.chpl,
and two affected .good files.

### CODE REVIEW

**Comment by cassella** [comment]
Hi, please particularly review:

- [ ] @bmcdonald3 - buildingandrunning.rst
- [ ] @DanilaFe - can-pass.h, Context.h
- [ ] @daviditen - OS.chpl
- [ ] @dlongnecke-cray - cleanups.cpp
- [ ] @mppf - QUICKSTART.rst, functionResolution.cpp
- [ ] @stonea  - gpu-cuda.c, chpl-init.c, Types.chpl, functionResolution.cpp

File: doc/rst/mason-packages/guide/buildingandrunning.rst [hunk 1]
```diff
@@ -10,7 +10,7 @@ When invoked, ``mason build [ options ]`` will do the following:
     - Run update to make sure any manual manifest edits are reflected in the dependency code.
     - Build ``MyPackage.chpl`` in the ``src/`` directory.
     - All packages are compiled into binaries and placed into ``target/``
-    - To forward options to the Chapel compiler(``chpl``), seperate them with a double dash
+    - To forward options to the Chapel compiler(``chpl``), separate them with a double dash
```
Comment (bmcdonald3) [inline doc/rst/mason-packages/guide/buildingandrunning.rst:13]: Good catch! Thanks! (Can you tell we use emacs/vim and don't have spell check 😄)

**Review by bmcdonald3 — APPROVED** [review]

**Comment by mppf** [comment]
These changes look good to me. They will need to wait until after the 1.28 release, though.

**Comment by daviditen** [comment]
They all look good to me for a post-release merge too. Thanks!

**Review by stonea — APPROVED** [review]

**Review by DanilaFe — APPROVED** [review]

File: modules/standard/Types.chpl [hunk 2]
```diff
@@ -1084,7 +1084,7 @@ proc isProperSubtype(type sub, type super) param {
 
 /* :returns: isProperSubtype(a,b) */
 pragma "docs only"
-deprecated "< operator is deprecated to compare types, use isPropersubtype instead"
```
[the remark this answers is not in the document]
Comment (cassella) [reply modules/standard/Types.chpl:1087]: I can do that, but should the docs-only operators be marked with `@unstable` instead?

[the remark this answers is not in the document]
Comment (stonea) [reply modules/standard/Types.chpl:1087]: > I can do that, but should the docs-only operators be marked with @unstable instead?

Yes, that would be best. I would add something like:

`@unstable "< operator is unstable when comparing types; consider using isSubtype() as a stable alternative"`

And:

`@unstable "<= operator is unstable when comparing types; consider using isProperSubtype() as a stable alternative"`

And similarly for `>` and `>=`. But feel free to tweak the wording.

Thanks!

[the remark this answers is not in the document]
Comment (cassella) [reply modules/standard/Types.chpl:1087]: Hmm.  For `<` and `<=`, that's easy enough (but I think `<` would go with `isProperSubtype`).  But I think there's more editorial decision-making for e.g. `>`:  `using isSubtype` or to spell out `using !isSubtype`, or take advantage of this being a doc-only warning to refer to the formals e.g. `operator >(a, b)` -> `using isProperSubtype(b, a)`. I think that's getting beyond this typo fix PR.

Additionally, the `isSubtype` and `isProperSubtype` doc comments a few lines earlier in the file suggest using the operators as an alternative to calling the functions, which should probably also be updated.

So I think I'd like to leave it for you subsequently as you suggested.  

FWIW, I did try out the docs build with the `@unstable` syntax, and it formats fine:

Screenshot from 2022-09-12 23-28-11

**Comment by cassella** [comment]
@dlongnecke-cray  Could you have a look?

**Review by dlongnecke-cray — APPROVED** [review]
Sorry for being the blocker here.

**Pull request merged**